engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -124,7 +153,727 @@ public static StramElement fromValue(String value) {
     }
 
   }
-  
+
+  /**
+   * This is an enum which represents a type of configuration.
+   */
+  protected enum ConfElement
```

Comment:
Is there a reason this is defined as an Enum instead of a Class? I feel that it would be much easier to work with as a class.